### PR TITLE
Feature/atom image

### PR DESCRIPTION
**Pull request by juanmaguitar**
Development of `AtomImage` component according to the specifications available at: https://paper.dropbox.com/doc/SUI-Image-Cv0S94htbykeN8cRkilPV

### CODE REVIEW

File: components/atom/image/src/index.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  &-figure {
+    background-color: $bgc-atom-image;
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: cover;
+    margin: 0;
+  }
+
+  &--placeholder.is-loading &-figure {
+    filter: blur(5px);
```
Comment (pachibadilla-scmspain) [inline components/atom/image/src/index.scss]: Am I not sure if it's really browser compatible; you could use a mixin to fallback a wide browser compatibiity solution ;-)

File: demo/atom/image/playground [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+return (
+  <div>
+    <h1>AtomImage</h1>
+    <ul style={listImageStyles}>
+      <li style={listItem}>
+        <div style={containerImageStyles}>
+          <AtomImage
+            style={{ border: '1px solid' }}
```
Comment (kikoruiz) [inline demo/atom/image/playground]: Do we really want to leave the door open to do this? Not sure if it's a good practice to leave "free" props opened to the public API of the component, regarding the following line: https://github.com/SUI-Components/sui-components/blob/8c3b74012d467547e0110ad19e53811dddec463c/components/atom/image/src/index.js#L66

A part from this, I think it's a very nice feature we got with this component 😉

File: components/atom/image/src/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  alt: PropTypes.string.isRequired,
+
+  /** Image displayed (blurred) while the final image is being loaded */
+  placeholder: PropTypes.string,
+
+  /** Image (wireframe, skeleton) displayed (not blurred) while the final image is being loaded */
+  skeleton: PropTypes.string,
+
+  /** Styles passed to the element that displays the skeleton as background */
+  bgStyles: PropTypes.object,
```
Comment (kikoruiz) [inline components/atom/image/src/index.js]: I would prefer to only style components via classnames, but if these styles one only affect to the skeleton/placeholder, maybe I'd switch this prop name to something more specific.

**Review by carlosvillu — APPROVED** [review]
👍

File: components/atom/image/src/index.scss [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  &.is-loaded &-spinner { 
+    display: none;
+  }
+
+  &-image {
+    backface-visibility: hidden;
+    height: 100%;
+    object-fit: cover;
+    transform: translateZ(0);
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/atom/image/src/index.scss]: I'm going to remove these 2 lines: 
```
backface-visibility: hidden;
transform: translateZ(0);
```

It was a fix i added for some glitch with the image when using `transition` with `filter`. But i'm testing this again and seems to be fine without these properties.

Thanks for catching that!

File: components/atom/image/src/index.scss (hunk 1 above)
Comment (JoanClaret) [reply components/atom/image/src/index.scss]: looks like pretty compatible, isn't it?
https://caniuse.com/#search=filter%3A%20blur

Comment (juanmaguitar) [reply components/atom/image/src/index.scss]: This `blur` effect is not part of the specification (https://paper.dropbox.com/doc/SUI-Image-Cv0S94htbykeN8cRkilPV). 

It's just a nice add that will work in most of the supported browsers according to this list: https://github.schibsted.io/scmspain/frontend-all-wiki/blob/on-boarding-guide/wiki/Components/new-component.md#browser-support

If this is not supported in some browser the final effect will still work according to the specs.

File: components/atom/image/src/index.js [hunk 5]
```diff
@@ -28,16 +28,25 @@ class AtomImage extends Component {
...
-      placeholder && `${BASE_CLASS}--placeholder`,
       `is-${loading ? 'loading' : 'loaded'}`,
       error && `is-error`
     )
   }
 
+  get classNamesFigure () {
+    const { placeholder, skeleton } = this.props
+    const BASE_CLASS_FIGURE = `${BASE_CLASS}-figure`
+    return cx(
```
Comment (kikoruiz) [inline components/atom/image/src/index.js]: 👍

**Review by kikoruiz — APPROVED** [review]

File: components/atom/image/src/defaults/ImageNotFoundIcon.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+          id='path-1'
+        />
+      </g>
+    </svg>
+  )
+}
+
+export default ImageNotFoundIcon
```
Comment (JoanClaret) [inline components/atom/image/src/defaults/ImageNotFoundIcon.js:11]: In other components, we use to import a SUI-icon as a default icon
https://github.com/SUI-Components/svg-iconset

there you have some examples: 
https://github.com/SUI-Components/sui-components/blob/master/components/breadcrumb/basic/src/index.js#L5
https://github.com/SUI-Components/sui-components/blob/master/components/molecule/notification/src/index.js#L4

To be more consistent, i would use this way.
The problem is that we does not have the icon you need in this library, so let's merge it and we'll discuss about how to proceed in a future similar situations.

right?

**Review by JoanClaret — APPROVED** [review]

File: components/atom/image/src/index.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  { className, icon: Icon, text } // eslint-disable-line react/prop-types
+) => (
+  <div className={className}>
+    {Icon && <Icon />}
+    <p>{text}</p>
+  </div>
+)
+
+class AtomImage extends Component {
+  constructor (props) {
```
Comment (midudev) [inline components/atom/image/src/index.js]: You could use the short form.

```
state = {
  loading: true,
  error: false
}
```

**Review by midudev — COMMENTED** [review]
Looks awesome man! :) Good job! 👏👏👏

File: components/atom/image/src/defaults/ImageNotFoundIcon.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+const ImageNotFoundIcon = () => {
+  return (
+    <svg width='40px' height='40px' viewBox='0 0 40 40'>
+      <g>
+        <path
```
Comment (midudev) [inline components/atom/image/src/defaults/ImageNotFoundIcon.js]: This code is unoptimized. You should use SVGOM in order to get the optimized output.

Comment (midudev) [reply components/atom/image/src/defaults/ImageNotFoundIcon.js]:
```
<svg width="40" height="40"><path d="M6.458 33.333c-1.72 0-3.125-1.382-3.125-3.077v-16.41c0-1.694 1.404-3.077 3.125-3.077h5.209l-.88.477 2.357-3.661a2.073 2.073 0 0 1 1.752-.918h9.791c.636 0 1.245.3 1.667.82l2.813 3.692-.834-.41h5.209c1.72 0 3.125 1.383 3.125 3.077v16.41c0 1.695-1.404 3.077-3.125 3.077H6.458zm0-2.05h27.084a1.04 1.04 0 0 0 1.041-1.027v-16.41a1.04 1.04 0 0 0-1.041-1.025h-5.209c-.328 0-.636-.152-.833-.41l-2.793-3.668c-.014-.017-.03-.025-.02-.025h-9.791c-.027 0-.01-.009-.006-.015l-2.344 3.64a1.046 1.046 0 0 1-.88.478H6.459a1.04 1.04 0 0 0-1.041 1.025v16.41a1.04 1.04 0 0 0 1.041 1.026zM20 29.23c-4.602 0-8.333-3.674-8.333-8.205 0-4.532 3.73-8.205 8.333-8.205 4.602 0 8.333 3.673 8.333 8.205 0 4.531-3.73 8.205-8.333 8.205zm0-2.052c3.452 0 6.25-2.755 6.25-6.153 0-3.399-2.798-6.154-6.25-6.154s-6.25 2.755-6.25 6.154c0 3.398 2.798 6.153 6.25 6.153zm-3.939-4.812a4.049 4.049 0 0 1 .751-3.984L3.66 5.23A1.111 1.111 0 0 1 5.23 3.66L18.7 17.127c.409-.132.847-.204 1.301-.204.454 0 .892.072 1.301.204L34.77 3.66A1.111 1.111 0 1 1 36.34 5.23L23.188 18.384a4.049 4.049 0 0 1 .75 3.983L36.342 34.77a1.111 1.111 0 0 1-1.571 1.571L22.634 24.205a4.19 4.19 0 0 1-2.634.923 4.19 4.19 0 0 1-2.634-.923L5.23 36.34A1.111 1.111 0 0 1 3.66 34.77L16.06 22.367zm14.356-5.444c-.576 0-1.042-.46-1.042-1.026 0-.566.466-1.025 1.042-1.025h1.041c.576 0 1.042.459 1.042 1.025 0 .567-.466 1.026-1.042 1.026h-1.041z"/></svg>
```

**Pull request merged**